packages/react/src/components/dialog/Dialog.tsx

```diff
@@ -0,0 +1,144 @@
+import React, { ReactNode, ReactNodeArray, useEffect, RefObject } from 'react';
+import ReactDOM from 'react-dom';
+
+// import core base styles
+import 'hds-core';
+import styles from './Dialog.module.scss';
+import classNames from '../../utils/classNames';
+import { useTheme } from '../../hooks/useTheme';
+import { IconCross } from '../../icons';
+import { DialogActionButtons } from './dialogActionButtons/DialogActionButtons';
+import { DialogHeader } from './dialogHeader/DialogHeader';
+import { DialogContent } from './dialogContent/DialogContent';
+
+export interface DialogCustomTheme {
+  '--accent-line'?: string;
+  '--color-overlay'?: string;
+  '--width'?: string;
+}
+
+export type DialogProps = React.PropsWithChildren<{
+  /**
+   * The id of the dialog element.
+   */
+  id: string;
+  /**
+   * The id of the heading element.
+   */
+  'aria-labelledby': string;
+  /**
+   * The id of the content description element which is usually a paragraph. The paragraph should be part of the content.
+   */
+  'aria-describedby'?: string;
+  /**
+   * When `true`, dialog is visible,
+   */
+  isOpen: boolean;
+  /**
+   * A function to control the isOpen property.
+   */
+  close?: () => void;
+  /**
+   * The id of the element which will get focus after the dialog is closed.
+   */
+  focusAfterCloseId?: string;
+  /**
+   * Custom theme styles
+   */
+  theme?: DialogCustomTheme;
+  /**
+   * Additional class names to apply to the dialog.
+   */
+  className?: string;
+  /**
+   * Children to render inside the dialog element.
+   */
+  children: ReactNode | ReactNodeArray;
+}>;
+
+export const Dialog = ({ id, isOpen, children, close, className, focusAfterCloseId, theme, ...props }: DialogProps) => {
+  const customThemeClass = useTheme<DialogCustomTheme>(styles.dialogContainer, theme);
+  const dialogRef: RefObject<HTMLInputElement> = React.createRef();
+
+  const { 'aria-labelledby': ariaLabelledby, 'aria-describedby': ariaDescribedby } = props;
+
+  const onKeyDown = (event: KeyboardEvent): void => {
```

Comment:
Small improvement to readability: We could maybe move this closer to the useEffect function where it is used? I see that setFocusAfterClose is also applied inside the useEffect though. Could this function come after the useEffect? There might be some lint rule that does not allow it though...

Reply:
I'll fix.